sphinx/source/docs/dev_guide/server.rst

```diff
@@ -31,9 +31,10 @@ widgets, or other content it desires. The application can also set up
 callbacks, to run periodically or to run when the document changes.
 
 Applications are represented by the ``Application`` class. This class is
-little more than a list of ``Handler`` instances. Handlers can be created
-in lots of ways; from JSON files, from Python functions, from Python files,
-and perhaps many more ways in the future.
+contains list of ``Handler`` instances and optional metainformation. Handlers
+can be created in lots of ways; from JSON files, from Python functions, from
+Python files, and perhaps many more ways in the future.  The optional metadata
+is available as a json blob via the ``/metadata`` endpoint.
```

Comment:
probably worth giving an example JSON output, with the fixed `url` and `data` fields, to demonstrate where exactly the user supplied metadata goes (in the `data` value)